### ISSUE TITLE
"No Sonos zone found"

### ISSUE THREAD

**Issue by raoulteeuwen** [janbar/noson-app#127]
Hello. Thank you for this project. I have a Sonos One. And try to use it with Kubuntu 19.10. When i install and run Noson, i get told "No Sonos zone found". In the Sonos app i can't find reference to a Zone, only to a room (i have one)? Am i missing something?

**Comment by janbar** [comment]
Hi, probably you have enabled firewall. Disable it and all should be ok.

**Comment by raoulteeuwen** [comment]
Did some more googl-ing. Found a message about someone sharing they found when the laptop/pc is connected to another AP/device in the LAN, it caused problems. I also have more than 1 AP. Switched off wifi so both my laptop and the Sonos One connect to the same AP > problem solved. Now only to find why the extra hop causes problems while uPNP is on, no FW afaik...

**Comment by janbar** [comment]
If you have more than one AP, you have more than one network enabled. The SSDP discovery cannot work in this context because this protocol use a broadcast packet. The broadcast request is send only to the first interface. If your are lucky it can work. Else I manage this situation by this way : 

In some circonstances your network couldn't allow broadcast/multicast traffic. To workaround the limitation you can configure the application by specifying the device endpoint.

Launch the Noson application from command line as follows.

    noson-app --deviceurl=http://<ipaddress>:1400 (the setting will be persistent).

Some SONOS devices could listen on port 3400 instead 1400.

Basically you have to launch the app one time using the command line with the parameter --deviceurl to advice the app you want to bypass the SSP discovery and force to use the specified url of your Sonos device: something like http:// IP : 1400 or 3400. If the connection succeed then the setting is stored.

Hope it can help.

**Comment by raoulteeuwen** [comment]
Thank you @janbar , that works!

**Comment by cpsyctc2** [comment]
Can confirm that this solved the same problem for me: ran:
`noson --deviceurl=http://<ipaddress>:1400`
I had to change "noson-app" to "noson" as I got a "not found" for noson-app and on my machine the port is 1400 and I had to do a bit of network mapping to find that my sonos speaker, a sonos 3, was sitting on 192.168.1.11. For what it's worth, on Ubuntu 22.04.3 LTS noson 5.6.0 (libnoson 2.11.2).  Many thanks for noson and this fix.
